mypy/checkexpr.py

```diff
@@ -2211,6 +2213,13 @@ def accept(self,
         self.type_context.pop()
         assert typ is not None
         self.chk.store_type(node, typ)
+
+        if (disallow_any and
+                not self.chk.is_stub and
+                has_any_type(typ) and
+                'expr' in self.chk.options.disallow_any):
```

Comment:
Nit: The last conditional should be first here.  It's standard (in this codebase at least) for conditionals involving flags for the flag to be checked first.  (And I think that makes the most conceptual sense, though I'm not sure I can well explain why.)